src/partition/ReceivedBoundingBox.hpp

```diff
@@ -0,0 +1,97 @@
+#pragma once
+
+#include "Partition.hpp"
+#include "logging/Logger.hpp"
+#include <vector>
+#include "mesh/Vertex.hpp"
+#include "mesh/Mesh.hpp"
+
+
+// Forward delcration to friend the boost test struct
+
+namespace precice {
+namespace partition {
+/**
+ * @brief A partition that is computed from a mesh received from another participant.
+ *
+ * A mesh is received by the master rank and re-partitioned among all slave ranks.
+ * Afterwards necessary distribution data structures are set up.
+ */
+class ReceivedBoundingBox : public Partition
+{
+public:
+
+  /// Defines the typ of geometric filter used
+  enum GeometricFilter {
```

Comment:
Please use a [scoped enum](https://en.cppreference.com/w/cpp/language/enum).
```suggestion
  enum class GeometricFilter {
```

Reply:
beyond this PR, on the list for final refactoring work package